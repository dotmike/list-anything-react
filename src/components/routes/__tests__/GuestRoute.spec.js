import configureStore from "redux-mock-store";

import ConnectedGuestRoute, { UnconnectedGuestRoute } from "../GuestRoute";

const setup = (propOverrides = {}) => {
  const props = {
    component: () => "Testando...",
    isAuthenticated: false,
    ...propOverrides
  };

  return {
    props,
    connectedWrapperShallow: wrapperShallow(ConnectedGuestRoute, props, true),
    // Lembrar de sempre chamar `unmount` quando usar esse wrapper \/
    wrapperMount: wrapperMount(UnconnectedGuestRoute, props, true)
  };
};

describe("ConnectedGuestRoute", () => {
  const mockStore = configureStore();
  const state = {};

  it("renders correctly", () => {
    const { connectedWrapperShallow: wrapper } = setup({
      store: mockStore(state)
    });
    expect(wrapper()).toMatchSnapshot();
  });
});

describe("UnconnectedGuestRoute", () => {
  it("renders correctly when `isAuthenticated` is false", () => {
    const { wrapperMount: wrapper } = setup({ isAuthenticated: false });
    expect(wrapper()).toMatchSnapshot();

    wrapper().unmount();
  });

  it("renders correctly when `isAuthenticated` is true", () => {
    const { wrapperMount: wrapper } = setup({ isAuthenticated: true });
    expect(wrapper()).toMatchSnapshot();

    wrapper().unmount();
  });
});
