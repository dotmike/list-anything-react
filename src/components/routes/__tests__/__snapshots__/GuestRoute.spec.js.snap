// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedGuestRoute renders correctly 1`] = `
<GuestRoute
  component={[Function]}
  dispatch={[Function]}
  isAuthenticated={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`UnconnectedGuestRoute renders correctly when \`isAuthenticated\` is false 1`] = `
<GuestRoute
  component={[Function]}
  isAuthenticated={false}
>
  <Route
    render={[Function]}
  >
    <component
      history={
        Object {
          "_listeners": Array [],
          "_notifyListeners": [Function],
          "action": "POP",
          "createHref": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "111111",
            "pathname": "/dashboard",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={
        Object {
          "hash": "",
          "key": "111111",
          "pathname": "/dashboard",
          "search": "",
        }
      }
      match={
        Object {
          "isExact": true,
          "params": Object {},
          "path": "/",
          "url": "/",
        }
      }
    >
      Testando...
    </component>
  </Route>
</GuestRoute>
`;

exports[`UnconnectedGuestRoute renders correctly when \`isAuthenticated\` is true 1`] = `
<GuestRoute
  component={[Function]}
  isAuthenticated={true}
>
  <Route
    render={[Function]}
  >
    <Redirect
      push={false}
      to="/dashboard"
    />
  </Route>
</GuestRoute>
`;
